Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -4663,5 +4755,17 @@ bool RenderBlockFlow::tryComputePreferredWidthsUsingModernPath(LayoutUnit& minLo
     return true;
 }
 
+LayoutUnit RenderBlockFlow::blockFormattingContextInFlowContentHeight() const
+{
+    ASSERT(createsBlockFormattingContext());
+    // For block layout we should just be able to check the height of the last in flow box
+    auto lastChild = lastInFlowChildBox();
```

Comment:
how does this work for simple cases like <div bfcroot>this does not have a render box only a render text</div>?

Reply:
Oh I think I misunderstood your previous comment. Yes I agree we should be able to get the content height from the inline content. I actually have this logic in my other patch (https://github.com/WebKit/WebKit/pull/9145/files#diff-8ddefc45aa5ce620250d06c1512ae9a263aaa9af625660ca8d50312e750c19f2) but I can bring it into this one as well.

Since I wanted to split it up into two patches (one to trim the floats in an IFC), I didn't add it here. Perhaps instead I can add a check `!hasChildrenInline` before calling `trimFloatBlockEndMargins` inside `RenderBlockFlow::layoutBlock` and then remove it in the IFC patch?

Reply:
I am sorry if my previous comment was misleading.
I think when you add a function called "foobar" it should do foobar, exactly what the name implies. At this point this function returns the height of the block level content and it should be named accordingly. When your upcoming patch changes the behavior, you should rename it. It helps to make sure nothing gets out-of-sync when reverts happen (i.e we don't end up with a function called blockFormattingContextInFlowContentHeight() with block level elements only and some other code starts calling it assuming it returns the height of the inflow content).

Reply:
Yeah that makes sense. I'll be more aware of making sure I changed function names and not only the logic associated with them if they get changed like this in future patches.

Reply:
Changed to `blockFormattingContextInFlowBlockLevelContentHeight`